File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.java
class Solution {
    public int appendCharacters(String s, String t) {
        int i = 0; // Pointer for string s
        int j = 0; // Pointer for string t
        while (i < s.length()) {
  
            while (i < s.length() && s.charAt(i) != t.charAt(j)) {
                i++;
            }
            if (i == s.length()) {
                return t.length() - j;
            }
            i++;
            j++;
            if (j == t.length()) {
                return 0;
            }
        }
        return t.length() - j;
    }
}

File: 2486-append-characters-to-string-to-make-subsequence/2486-append-characters-to-string-to-make-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {number}
 */
var appendCharacters = function(s, t) {
    let i = 0;
    let j = 0;
    
    while (i < s.length) {
        while (i < s.length && s.charAt(i) != t.charAt(j)) {
            i++;
        }
        if (i == s.length) {
            return t.length - j;
        }
        i++;
        j++;
        if (j == t.length) {
            return 0;
        }
    }
    return t.length - j;
};